app/renderer/components/frame/frame.js

```diff
@@ -112,58 +103,44 @@ class Frame extends React.Component {
     return !this.webview
   }
 
-  runInsecureContent () {
-    const activeSiteSettings = siteSettings.getSiteSettingsForHostPattern(this.props.allSiteSettings, this.origin)
-    return activeSiteSettings === undefined
-      ? false : activeSiteSettings.get('runInsecureContent')
-  }
-
   allowRunningWidevinePlugin (url) {
     if (!this.props.isWidevineEnabled) {
       return false
     }
-    const origin = url ? siteUtil.getOrigin(url) : this.origin
-    if (!origin) {
+    if (!this.props.origin) {
       return false
     }
     // Check for at least one CtP allowed on this origin
     if (!this.props.allSiteSettings) {
       return false
     }
-    const activeSiteSettings = siteSettings.getSiteSettingsForHostPattern(this.props.allSiteSettings, origin)
-    if (activeSiteSettings && typeof activeSiteSettings.get('widevine') === 'number') {
+    if (typeof this.props.widevine === 'number') {
       return true
     }
     return false
   }
 
-  expireContentSettings (origin) {
+  expireContentSettings (props) {
     // Expired Flash settings should be deleted when the webview is
     // navigated or closed. Same for NoScript's allow-once option.
-    const activeSiteSettings = siteSettings.getSiteSettingsForHostPattern(this.props.allSiteSettings, origin)
-    if (!activeSiteSettings) {
-      return
-    }
-    if (typeof activeSiteSettings.get('flash') === 'number') {
-      if (activeSiteSettings.get('flash') < Date.now()) {
-        appActions.removeSiteSetting(origin, 'flash', this.props.isPrivate)
+    if (typeof props.flash === 'number') {
+      if (props.flash < Date.now()) {
+        appActions.removeSiteSetting(props.origin, 'flash', props.isPrivate)
       }
     }
-    if (activeSiteSettings.get('widevine') === 0) {
-      appActions.removeSiteSetting(origin, 'widevine', this.props.isPrivate)
+    if (props.widevine === 0) {
+      appActions.removeSiteSetting(props.origin, 'widevine', props.isPrivate)
     }
-    if (activeSiteSettings.get('noScript') === 0) {
-      appActions.removeSiteSetting(origin, 'noScript', this.props.isPrivate)
+    if (props.noScript === 0) {
+      appActions.removeSiteSetting(props.origin, 'noScript', props.isPrivate)
     }
-    const noScriptExceptions = activeSiteSettings.get('noScriptExceptions')
-    if (noScriptExceptions) {
-      appActions.noScriptExceptionsAdded(origin,
-        noScriptExceptions.map(value => value === 0 ? false : value))
+    if (props.noScriptExceptions) {
+      appActions.noScriptExceptionsAdded(props.origin, this.props.noScriptExceptions.filter((value, host) => value !== 0 ? false : value))
```

Comment:
also i think it should be props.noScriptExceptions instead of this.props.noScriptExceptions